docs/src/adr/019adr-changelog-management.md

```diff
@@ -0,0 +1,182 @@
+
+This process creates a race condition over `UNRELEASED.md` between any
+concurrent pull requests. This consequently results in developers constantly
+having to resolve merge conflicts. This busy-work slows down development and
+adds no value.
+
+## Options
+
+All of the following options would resolve the problem of merge conflicts, and
+could be integrated into our existing automated deployment pipeline.
+
+However they have different costs w/r/t dependencies, processes, and learning
+curve.
+
+### 1. Use one of the automated changelog generators
+
+There is a [superabundance](https://github.com/search?q=changelog+auto) of tools
+for automatically generating changelogs. These tools extract changelogs from
+various different sources such as commit messages, issues, or pull requests.
+    
+#### Pros
+
+- Requires no new development
+- Would eliminate the need to add changelog entries in a separate file
+
+#### Cons
+
+- Developers have to learn the annotation format required by the tool, and they
+  have to remember to deploy it in their commits/PRs/issues.
+- Infrastructure maintainers have to learn how to configure and maintain the tool.
+- We pickup a dependency external to both the company and our project
+- Changelogs are meant to communicate a very specific kind of information to a
+  specific audience. Git commit messages, pull requests, and issues are for
+  communicating quite different information to an entirely different audience.
+  Requiring that these communication channels be overloaded risks degrading the
+  quality of communications in all the conflated channels.
+  - However, it's possible we could work out some conventions to add empty
+    commits or designate special PR labels to work around this interference?
+
+
+### 2. Use Unclog
+
+@thanethompson has written the [unclog][] utility, which is used in both
+[informalsystems/tendermint-rs][] and [informalsystems/ibc-rs][]. The tool
+stores changelog entries in a directory structure, with each entry in its own
+file, ensuring merge conflicts can't arise. 
+
+Entries are added via the CLI. E.g.:
+
+```sh
+unclog add --id some-new-feature \
+  --issue 23 \
+  --section breaking-changes \
+  --message "Some *new* feature"
+```
+
+If the `--messages` argument is omitted, it will open your configured editor for
+authoring the message.
+
+The tool has subcommands to generate and update the `CHANGES.md`, and it supports
+a variety of options via its TOML configuration.
+
+[unclog]: https://github.com/informalsystems/unclog
+[informalsystems/tendermint-rs]: https://github.com/informalsystems/tendermint-rs
+[informalsystems/ibc-rs]: https://github.com/informalsystems/ibc-rs
+
+#### Pros
+
+- Requires no new development
+- It is developed internally to informal
+
+#### Cons
+
+- Adds cognitive and procedural overhead to adding changelog entries
+- Would add a rust dependency to our devenv
+- Requires external contributors to either install a CLI tool or figure out a
```

Comment:
I don't know much about the rust stack; is there a way to keep the installation very simple, e.g. precompiled binaries for common platforms?

Reply:
That would keep the installation simple for those platforms, but this adds even more overhead for the setup, since no such precompiled are currently available. afaiu, rust libs dynamically link against glibc, so it doesn't provide the same sort of highly portable binaries we'd get from Go, or a JAR file (modulo JVM req).

We could package it for each platform, but that's an even heavier lift.

For the sake of reducing setup speed in our CI, we could precompile a binary for the particular architecture we use (we do this for emojitsu). But this won't help developers or external contributors much.

Still, installing from cargo is not difficult, it's just one extra step.

Moreover, as @konnov has pointed out, it should be possible to just maintain the entries by hand in the directory structure, as we'd do any how.

The format is a bit finicky (for some reason requiring an issue number to start every file name) and they have to be formatted as a single bullet point, but it's not too terribly confusing if we document it clearly.

Reply:
Okay, precompiling probably seems not very beneficial, thanks for the insight.

Reply:
It is a good thing to consider if we find a CI slowdown as a result of this dependency! It would also help users, of course. It's just that maintaining packages for bunch of different systems is quite a bit of work (afaik) and since it's not already already done, that work would probably fall on us.

We **could** probably drop this dependency into our nix flake configuration. That might be the best option here from a devenv perspective. I'll include a note about that.